services-directions-models/src/main/java/com/mapbox/api/directions/v5/utils/ParseUtils.java

```diff
@@ -17,189 +16,201 @@
-          result.add(null);
-        } else {
-          result.add(str);
-        }
-      }
-    }
-
-    return result;
+    return parseToList(separator, original, STRING_PARSER);
   }
 
   /**
-   * Parse a String to a list of Points.
+   * Parse a semicolon separated String to a list of Points.
+   * If separation finds empty strings, those will be added as nulls to the resulting list.
+   * <p>
+   * Elements of the string should have 2 comma separated double values,
+   * longitude and latitude, for example:
+   * "10.1,47.3;33.09,79.111"
    *
    * @param original an original String.
-   * @return List of Points
+   * @return List of Points or null if the original string is null
    */
   @Nullable
   public static List<Point> parseToPoints(@Nullable String original) {
-    if (original == null) {
-      return null;
-    }
-
-    List<Point> points = new ArrayList<>();
-    String[] targets = original.split(SEMICOLON, -1);
-    for (String target : targets) {
-      if (target != null) {
-        if (target.isEmpty()) {
-          points.add(null);
-        } else {
-          String[] point = target.split(COMMA);
-          points.add(Point.fromLngLat(Double.valueOf(point[0]), Double.valueOf(point[1])));
-        }
-      }
-    }
-
-    return points;
+    return parseToList(SEMICOLON, original, POINT_PARSER);
   }
 
   /**
-   * Parse a String to a list of Points.
+   * Parse a semicolon separated String to a list of Doubles.
+   * If separation finds empty strings, those will be added as nulls to the resulting list.
    *
    * @param original an original String.
-   * @return List of Doubles
+   * @return List of Doubles or null if the original string is null
    */
   @Nullable
   public static List<Double> parseToDoubles(@Nullable String original) {
-    if (original == null) {
-      return null;
-    }
-
-    List<Double> doubles = new ArrayList<>();
-    String[] strings = original.split(SEMICOLON, -1);
-    for (String str : strings) {
-      if (str != null) {
-        if (str.isEmpty()) {
-          doubles.add(null);
-        } else if (str.equals(UNLIMITED)) {
-          doubles.add(Double.POSITIVE_INFINITY);
-        } else {
-          doubles.add(Double.valueOf(str));
-        }
-      }
-    }
-
-    return doubles;
+    return parseToList(SEMICOLON, original, DOUBLE_PARSER);
   }
 
   /**
-   * Parse a String to a list of list of Doubles.
+   * Parse a semicolon separated String to a list of lists of doubles.
+   * If separation finds empty strings, those will be added as nulls to the resulting list.
+   * <p>
+   * Elements of the string should have a comma separated double values, for example:
+   * ";;10.1,47.3,33.09,79.111;84.45,45.4;"
    *
    * @param original an original String.
-   * @return List of List of Doubles
+   * @return List of List of Doubles or null if the original string is null
    */
   @Nullable
   public static List<List<Double>> parseToListOfListOfDoubles(@Nullable String original) {
```

Comment:
I'd restrict `List<List<Double>>` at all, and replace it with `List<Point>` in all places of mapbox-java

Reply:
This is actually used for `bearings` parameter which is a pair of doubles. Those are different than points and unfortunately, we can't use the same method.

Reply:
👍 my bad

Should we add a wrapper for bearings? that list of beings looks like `List<Bearings>`. It would be good for documentation as well: we can add javadocs for each value of `Beraings`